src/Observers.jl

```diff
@@ -41,11 +43,38 @@ Base.copy(observer::Observer) =
 results(observer::Observer, obsname::String) = 
   last(observer[obsname])
 
+"""
+    update!(obs::Observer, args...; kwargs...)
+
+Update the observer by executing the functions in it.
+"""
 function update!(obs::Observer, args...; kwargs...)
+  # loop over the functions
   for (k, v) in obs
     obs_k = obs[k]
+    # if a function is defined
     if !isnothing(obs_k.f)
-      push!(obs_k.results, obs_k.f(args...; kwargs...))
+      # collect the types of each positional argument being passed into the observer
+      args_types = typeof.(args)
+      # initialize the result to catch a method not being found
+      result = MissingMethod() 
+      # loop over the sequences of possible positional arguments
+      for i in 0:length(args_types)
+        tlist = args_types[1:i]
+        # if a function with argument matching tlist exists:
+        if hasmethod(obs_k.f, tlist)
+          # check the kwargs of such function
+          kwargs_list = Base.kwarg_decl(which(obs_k.f, tlist))
+          # remove the kwargs... from the kwargs
+          filter!(x -> x ≠ Symbol("kwargs..."), kwargs_list)
+          # execute the function
+          result = isempty(kwargs_list) ? obs_k.f(args[1:i]...) : 
+                                          obs_k.f(args[1:i]...; (kwargs_list .=> [kwargs[κ] for κ in kwargs_list])...)
```

Comment:
You should be able to use `(kwargs[κ] for κ in kwargs_list)` which won't allocate a `Vector`.